### DISCUSSION TITLE
Improve music player

### DISCUSSION THREAD

**Discussion by Bytezz** [libre-tube/LibreTube#6654] [Ideas]
### Describe your suggested feature

Hi and thanks for this nice software.  
I would like to suggest a bunch of points to improve LibreTube as a music player.  
I can edit/remove points considered out of scope and keep this ticket open to track overall progress.

- [ ] Swiping down to minimize audio player #4734
- [ ] Swiping left/right to play next/previous song
- [ ] Remove " - Topic" in songs (if in music player mode)
- [ ] Better auto queue (more on this later)
- [ ] Buffer first seconds of next song (if in music player mode)
- [ ] Get lyrics of songs

I'm on version 0.25.1 (56) from F-Droid.
Please let me know if someone needs testing.

### Other details

About auto queue, often in queue after a YouTube Music song its put the video of that song and or multiple remixes of that same song and sometimes its put only the video version while the song it's available in YouTube Music.

For the lyrics a free API has to be used, like https://lyrist.vercel.app/ or https://lyrics.ovh/ but synced lyrics would be better.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
The auto queue and lyrics would require the usage of additional APIs, which is not planned.

I don't personally think the other issues are really useful to the average user honestly.

**Reply by Bytezz** [reply]
> would require the usage of additional APIs

Should be enough to check for title+artist of an item in queue if there is the YouTube Music version of it and replace it with that or remove from the queue if already present. This way would remove duplicates in queue.  
It's a thing I stumble upon very often.

> I don't personally think the other issues are really useful to the average user honestly.

Probably not, since this is primarily a video player, these are just bells and whistles around the music player with low priority.

I'm sorry if giving a list of features like this could be seen as too much demanding and stressfull. There is of course no rush and these can be done even in the far future. I just wanted to know what is your position about these points, if relevant to the project and not out of scope.

Thank you.

**Reply by Bnyro** [reply]
It's been converted to a discussion to let other people give their opinion on it - if there's a high demand, it might be possible to implement some of the features sometime in the future, but in case you really want these I'd personally use an other existing YouTube music player for Android (there are lots of them on F-Droid) for specifically playing music with lyrics and better queuing.
